examples/api/lib/material/tabs/tab_bar.onFocusChange.dart

```diff
@@ -0,0 +1,60 @@
+import 'package:flutter/material.dart';
+
+/// Flutter code sample for [TabBar.onFocusChange].
+
+void main() => runApp(const TabBarApp());
+
+class TabBarApp extends StatelessWidget {
+  const TabBarApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(theme: ThemeData(useMaterial3: true), home: TabBarExample());
+  }
+}
+
+class TabBarExample extends StatefulWidget {
+  TabBarExample({super.key});
+
+  @override
+  State<TabBarExample> createState() => _TabBarExampleState();
+}
+
+class _TabBarExampleState extends State<TabBarExample> {
+  int? focusedIndex;
+
+  @override
+  Widget build(BuildContext context) {
+    return DefaultTabController(
+      initialIndex: 1,
+      length: 3,
+      child: Scaffold(
+        appBar: AppBar(
+          title: const Text('TabBar Sample'),
+          bottom: TabBar(
+            onFocusChange: (bool value, int index) {
+              setState(() {
+                focusedIndex = switch (value) {
+                  true => index,
+                  false => null,
```

Comment:
Do we need to worry about cases where false is called after true?
for example switch from focus from tab 1-> tab2
whether tabs 1 false may be called after tab 2 true

Reply:
If we don't we should probably document this on the api

Reply:
> Do we need to worry about cases where false is called after true?

I don't think so, I've tested it, I was curious about the same exact thing.

Reply:
Ah gotcha! Added!